src/Core/Util/SymfonyArrayFinder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+class SymfonyArrayFinder implements \ArrayAccess, \Countable
+{
+    /**
+     * @var array
+     */
+    private $array;
+    /**
+     * @var PropertyAccessor
+     */
+    private $propertyAccessor;
+
+    public function __construct(array $content = [])
+    {
+        $this->array = $content;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return count($this->array);
+    }
+
+    private function convertDotPathToArrayPath(string $dotPath): string
+    {
+        if ($dotPath === '[]') {
+            return '[0]';
+        }
+
+        $expl = explode('.', $dotPath);
+        $in = implode('][', $expl);
+
+        return '[' . $in . ']';
+    }
+
+    /**
+     * @param string $path
+     * @param mixed $default
+     *
+     * @return mixed|null
+     */
+    public function get(string $path = null, $default = null)
```

Comment:
This class reproduces the behavior of https://github.com/Shudrum/ArrayFinder/blob/master/ArrayFinder.php

Reply:
This isn't something you should write here but in the code. there is a phpdoc block for that :wink:

Reply:
I have written this class 100% myself. I follow the Shudrum ArrayFinder signature as I need to match it, but the implementation is 100% mine.

If we say "following code X signature / public API means it's a derivative work", then PrestaShop BO is slowly becoming a derivative work of Symfony 🤔  .

I aim to replace the old implementation by my own (which is why I mimic its behavior), not to use it.

Reply:
That being said, I add the comment inside my code if that makes you feel better 😄  .

Reply:
Done in https://github.com/PrestaShop/PrestaShop/pull/25611/commits/c0066eb57f7c1f3ed17205f820cc4a0a789c5073

Reply:
> "following code X signature / public API means it's a derivative work"

This was exactly the argument of Oracle in their lawsuit against Google for their Java implementation in Android. While the question of API copyright hasn't been completely answered, there's a good precedent for [fair use](https://arstechnica.com/tech-policy/2021/04/how-the-supreme-court-saved-the-software-industry-from-api-copyrights/).

Reply:
Done in https://github.com/matks/PrestaShop/commit/107e31a133288b35efeb0c6f3d72d5d2b73c8fa3